src/miner.cpp

```diff
@@ -726,7 +727,7 @@ void static BitcoinMiner()
                 // Update nNonce and nTime
                 pblock->nNonce = ArithToUint256(UintToArith256(pblock->nNonce) + 1);
                 UpdateTime(pblock, chainparams.GetConsensus(), pindexPrev);
-                if (chainparams.GetConsensus().fPowAllowMinDifficultyBlocks)
+                if (chainparams.GetConsensus().nPowAllowMinDifficultyBlocksFromHeight)
```

Comment:
It isn't clear how a `boost::optional` that holds 0 (which is the case for regtest) is coerced to a boolean, unless you pore over the Boost documentation &mdash; which isn't actually very clear on this point. Make it more explicitly a check that the `optional` is set:
```
if (chainparams.GetConsensus().nPowAllowMinDifficultyBlocksFromHeight != boost::none)
```

Reply:
Since the member starts with 'nPow' a reader might assume the member value is a number, just like`nPowMaxAdjustDown` and not realize it's actually a boost optional.  So perhaps also rename to `optPow` or optNPow` etc.?